src/components/OperationsList/index.js

```diff
@@ -91,89 +78,124 @@ const ShowMore = styled(Box).attrs({
   p: 4,
   color: 'wallet',
 })`
-  border-top: 1px solid ${p => p.theme.colors.fog};
   cursor: pointer;
   &:hover {
     text-decoration: underline;
   }
 `
 
+const AddressEllipsis = styled.div`
+  display: block;
+  flex-shrink: 1;
+  min-width: 20px;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+`
+
+const Address = ({ value }: { value: string }) => {
+  const addrSize = value.length / 2
+
+  const left = value.slice(0, 10)
+  const right = value.slice(-addrSize)
+  const middle = value.slice(10, -addrSize)
+
+  return (
+    <Box horizontal color="smoke" ff="Open Sans" fontSize={3}>
+      <div>{left}</div>
+      <AddressEllipsis>{middle}</AddressEllipsis>
+      <div>{right}</div>
+    </Box>
+  )
+}
+
 const Operation = ({
   account,
-  t,
+  counterValue,
+  counterValues,
+  minConfirmations,
   onAccountClick,
+  t,
   tx,
   withAccount,
-  minConfirmations,
 }: {
-  account: Account | null,
-  t: T,
+  account: Account,
+  counterValue: string,
+  counterValues: Object | null,
+  minConfirmations: number,
   onAccountClick?: Function,
+  t: T,
   tx: OperationType,
   withAccount?: boolean,
-  minConfirmations: number,
 }) => {
-  const acc = account || tx.account
+  const { unit } = account
   const time = moment(tx.receivedAt)
-  const Icon = getIconByCoinType(get(tx, 'account.currency.coinType'))
+  const Icon = getIconByCoinType(get(account, 'currency.coinType'))
```

Comment:
why not doing `account.currency.coinType` ? We should try to limit nb of usage of `get` to not lose the (flow) type

Reply:
Yup, you're right, I fix that